### ISSUE TITLE
Hex digests vs buffers

### ISSUE THREAD

**Issue by stephan83** [c-geek/merkle#9]
The current implementation stores the hashes as hexadecimal strings and computes parent hashes doing something like this:

```
parentHashHex = hash(leftHex + rightHex).toString('hex')
```

I'm not sure if this is a standard, but it means that the parent hash is encoding dependent.

To me, it makes more sense to use buffers and concatenate the binary data directly:

```
parentHashBuf = hash(Buffer.concat([leftBuf, rightBuf]))
```

This way, the computation does not rely on encoding. Since it decouples encoding, it also makes it possible to use other encodings such as base64 and still get the same parent hashes.

**Comment by c-geek** [comment]
Thanks for your comment.

Indeed, it seems more correct to use binary nodes & leaves instead of encoded data since binary is a more deep and common form for data.

However this would be a breaking change I guess. So, we could include this for a major version.

**Comment by stephan83** [comment]
Yes it is indeed a very breaking change, since it also results in different hashes. Maybe it would be better to make it optional and introduce it as a non-breaking change?

**Comment by c-geek** [comment]
Yes we can for current version, however I plan to make a v1.0.0 some day along with code cleaning, so we could make breaking changes that time.

**Comment by stephan83** [comment]
Great, it seems fine for 1.0.0, no need to introduce complexity in the current version. In the meantime I have a branch that does it that I can use for now. Let me know if you need help for 1.0.0.
